skrifa/src/outline/native_hint.rs

```diff
@@ -0,0 +1,147 @@
+    /// outline collection and settings.
+    pub fn reconfigure<'a>(
+        &mut self,
+        outlines: &OutlineGlyphCollection,
+        size: Size,
+        location: impl Into<LocationRef<'a>>,
+        mode: NativeHinting,
+    ) -> Result<(), DrawError> {
+        self.size = size;
+        self.coords.clear();
+        self.coords.extend_from_slice(location.into().coords());
+        self.mode = mode;
+        // Reuse memory if the font contains the same outline format
+        let current_kind = core::mem::replace(&mut self.kind, HinterKind::None);
+        match &outlines.kind {
+            OutlineCollectionKind::Glyf(_) => {
+                self.kind = HinterKind::Glyf();
+            }
+            OutlineCollectionKind::Cff(cff) => {
+                let mut subfonts = match current_kind {
+                    HinterKind::Cff(subfonts) => subfonts,
+                    _ => vec![],
+                };
+                subfonts.clear();
+                let size = size.ppem().unwrap_or_default();
+                for i in 0..cff.subfont_count() {
+                    subfonts.push(cff.subfont(i, size, &self.coords)?);
+                }
+                self.kind = HinterKind::Cff(subfonts);
+            }
+            OutlineCollectionKind::None => {}
+        }
+        Ok(())
+    }
+
+    pub(super) fn draw_hinted(
+        &self,
+        glyph: &OutlineGlyph,
+        memory: Option<&mut [u8]>,
+        pen: &mut impl OutlinePen,
+    ) -> Result<AdjustedMetrics, DrawError> {
+        let ppem = self.size.ppem().unwrap_or_default();
+        let coords = self.coords.as_slice();
+        match (&self.kind, &glyph.kind) {
+            (HinterKind::Glyf(..), OutlineKind::Glyf(glyf, outline)) => {
+                super::with_glyf_memory(outline, Hinting::Native, memory, |buf| {
+                    let mem = outline
+                        .memory_from_buffer(buf, Hinting::Native)
+                        .ok_or(DrawError::InsufficientMemory)?;
+                    let scaled_outline = glyf.scale(mem, outline, ppem, coords)?;
+                    scaled_outline.to_path(pen)?;
+                    Ok(AdjustedMetrics {
+                        has_overlaps: outline.has_overlaps,
+                        lsb: Some(scaled_outline.adjusted_lsb().to_f32()),
+                        advance_width: Some(scaled_outline.adjusted_advance_width().to_f32()),
+                    })
+                })
+            }
+            (HinterKind::Cff(subfonts), OutlineKind::Cff(cff, glyph_id, subfont_ix)) => {
+                let Some(subfont) = subfonts.get(*subfont_ix as usize) else {
+                    return Err(DrawError::NoSources);
+                };
+                cff.outline(subfont, *glyph_id, &self.coords, true, pen)?;
+                Ok(AdjustedMetrics::default())
+            }
+            _ => Err(DrawError::NoSources),
+        }
+    }
+}
+
+#[derive(Clone)]
+enum HinterKind {
+    None,
+    Glyf(),
```

Comment:
Why `()`?

Reply:
Useless but harmless? I probably added it as a placeholder for the upcoming TT hinting state 🤷🏻‍♂️